### PR TITLE
Implement `ReflectionClass#getTraitClassNames()`

### PR DESCRIPTION
**Pull request by staabm**
same motivation as in https://github.com/Roave/BetterReflection/pull/1359, but this time for traits:-)

refs https://github.com/rectorphp/rector/issues/8093#issuecomment-1657057947

### CODE REVIEW

File: test/unit/Reflection/ReflectionClassTest.php [hunk 1]
```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
...
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraits();
+    }
```
Comment (staabm) [inline test/unit/Reflection/ReflectionClassTest.php:1527]: these tests show, that the pre-PR existing API always throws and therefore does not allow to get the interface names, when the class-symbols are not autoloadable.

File: test/unit/Reflection/ReflectionClassTest.php [hunk 2]
```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
...
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        self::assertSame(
+            [],
+            $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitAliases(),
+        );
```
Comment (staabm) [inline test/unit/Reflection/ReflectionClassTest.php:1539]: `getTraitAliases()` does not require autoloadable symbols, therefore runs without exceptions

File: test/unit/Reflection/ReflectionClassTest.php [hunk 3]
```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
...
+
+    /** @return list<array{string, class-string, list<class-string>}> */
+    public static function getTraitClassNamesDataProvider(): array
+    {
+        return [
+            [
+                __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+                Fixture\ClassUsesUnknownTrait::class,
+                ['Roave\BetterReflectionTest\Fixture\UnknownTrait'],
+            ],
```
Comment (staabm) [inline test/unit/Reflection/ReflectionClassTest.php:1565]: this first test-case shows that we are now able to extract the trait names even if not autoloadable.

the remaining cases add coverage for a few more related cases

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @staabm!

**Pull request merged**